nautobot_chatops/dispatchers/base.py

```diff
@@ -37,6 +36,26 @@ def __init__(self, context=None):
         """Init this Dispatcher with the provided dict of contextual information (which will vary by app)."""
         self.context = context or {}
 
+    def _get_cache_key(self) -> str:
+        """Key generator for the cache, adding the plugin prefix name."""
+        # Using __file__ as a key customization within the cache
+        key_string = "-".join([__file__, self.context.get("user_name")])
+        return hashlib.md5(key_string.encode("utf-8")).hexdigest()  # nosec
+
+    @property
+    def session(self) -> dict:
+        """Return the session data stored for the user, that should be a dict."""
+        return cache.get(self._get_cache_key()) or {}
+
+    @session.setter
+    def session(self, value: dict):
+        """Set the session data for a user."""
+        cache.set(self._get_cache_key(), value, timeout=86400)
```

Comment:
Timeout should be configurable with a default.

Reply:
totally